src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StageChunkFinder.java

```diff
@@ -22,7 +26,19 @@ public boolean isStartInsideChunk() {
 
     @Override
     public boolean isChunkStart(@Nonnull FlowNode current, @CheckForNull FlowNode previous) {
-        return current.getAction(LabelAction.class) != null && current.getAction(ThreadNameAction.class) == null;
+        if ((current instanceof StepAtomNode || current instanceof StepStartNode) && !((((StepNode) current).getDescriptor()) instanceof StageStep.DescriptorImpl)) {
```

Comment:
Not sure what that means exactly, but anyway the Javadoc does claim to be specific to be `stage`s.